org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java

```diff
@@ -595,7 +595,7 @@ public static String toURI(ICompilationUnit cu) {
 	 * @return
 	 */
 	public static String getFileURI(IResource resource) {
-		return ResourceUtils.fixURI(resource.getRawLocationURI());
+		return ResourceUtils.fixURI(resource.getRawLocationURI() != null ? resource.getRawLocationURI() : resource.getLocationURI());

```

Comment:
I prefer the form `something == null? this : that`. So the CPU doesn't need the extra instruction to revert the outcome of the check. I'm old fashioned